src/sidebar/store/modules/frames.ts

```diff
@@ -163,7 +163,16 @@ const mainFrame = createSelector(
  */
 const defaultContentFrame = createSelector(
   (state: State) => state.frames,
-  frames => firstFrameWithoutId(frames) ?? frames[0] ?? null,
+  frames => {
+    const mainFrame = firstFrameWithoutId(frames);
+    if (mainFrame) {
+      return mainFrame;
+    } else if (frames.length > 0) {
+      return frames[0];
+    } else {
+      return null;
+    }
+  },
```

Comment:
I think the logic before was correct in practice, what was not correct is how TypeScript was inferring the type as `Frame` instead of `Frame | null`.

What we should probably consider is setting `noUncheckedIndexedAccess` as `true` (it's `false` by default).

See https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess and https://twitter.com/mattpocockuk/status/1510938191091712007?lang=en

Reply:
I see you already considered this https://github.com/hypothesis/client/pull/5819#discussion_r1324201995